### PR TITLE
8293167: Memory leak in JfrThreadSampler if stackdepth is larger than default (64)

### PR DESCRIPTION
**Pull request by mgronlun**
Greetings,

please help review this fix that address a memory leak in the JfrThreadSampler when using a stackdepth larger than default (64).

Please see the JIRA issue for details.

Testing: jdk_jfr, stress

Thanks
Markus

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8293167](https://bugs.openjdk.org/browse/JDK-8293167): Memory leak in JfrThreadSampler if stackdepth is larger than default (64)

### Reviewers
 * [Jaroslav Bachorik](https://openjdk.org/census#jbachorik) (@jbachorik - **Reviewer**) ⚠️ Review applies to [47c8af96](https://git.openjdk.org/jdk/pull/10129/files/47c8af96e833568532b2d84826a1e3443c180220)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10129/head:pull/10129` \
`$ git checkout pull/10129`

Update a local copy of the PR: \
`$ git checkout pull/10129` \
`$ git pull https://git.openjdk.org/jdk pull/10129/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10129`

View PR using the GUI difftool: \
`$ git pr show -t 10129`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10129.diff">https://git.openjdk.org/jdk/pull/10129.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/jfr/recorder/checkpoint/types/traceid/jfrTraceIdLoadBarrier.cpp [hunk 1]
```diff
@@ -25,54 +25,86 @@
...
 #include "jfr/utilities/jfrEpochQueue.inline.hpp"
 #include "runtime/javaThread.hpp"
 #include "runtime/mutexLocker.hpp"
+#include "utilities/powerOfTwo.hpp"
 
-// The queue instance used by the load barrier to enqueue tagged Klass'es.
-static JfrTraceIdKlassQueue* _klass_queue = NULL;
+// The queue instances are used by the load barrier to enqueue tagged Klass'es.
+static JfrTraceIdKlassQueue* _klass_queue = nullptr; // Generic for all Java threads.
+static JfrTraceIdKlassQueue* _sampler_klass_queue = nullptr; // Specialized for the Jfr Thread Sampler using use a larger buffer size.
```
Comment (jbachorik) [inline src/hotspot/share/jfr/recorder/checkpoint/types/traceid/jfrTraceIdLoadBarrier.cpp:37]:
```suggestion
static JfrTraceIdKlassQueue* _sampler_klass_queue = nullptr; // Specialized for the Jfr Thread Sampler using a larger buffer size.
```

**Review by jbachorik — APPROVED** [review]
Looking good.
I could find only one small error in a comment :)

File: src/hotspot/share/jfr/recorder/checkpoint/types/traceid/jfrTraceIdLoadBarrier.cpp (hunk 1 above)
Comment (mgronlun) [reply src/hotspot/share/jfr/recorder/checkpoint/types/traceid/jfrTraceIdLoadBarrier.cpp]: Thanks @jbachorik for noticing.

**Comment by mgronlun** [comment]
Thank you @jbachorik for reviewing.

**Pull request closed without merging**